frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.cyspec.ts

```diff
@@ -0,0 +1,57 @@
+import { testName } from '../../utils/common';
+import { listPage } from '../../views/list-page';
+import { modal } from '../../views/modal';
+
+describe('Namespace', () => {
+  before(() => {
+    cy.login();
+    cy.createProject(testName);
+  });
+  after(() => {
+    cy.deleteProject(testName);
+    cy.logout();
+  });
+
+  const newName = `${testName}-ns`;
+
+  it('Create, Retrieve (list), Update (labels)', () => {
+    const labelAppFrontend = 'app=frontend';
+
+    cy.log('Test Namespace list page');
+    cy.visit('/k8s/cluster/namespaces');
+    listPage.rows.shouldNotExist(newName);
+    listPage.filter.byName(testName);
+    listPage.rows.shouldExist(testName); // created via cy.createProject(testName) above
+
+    cy.log('Creates the Namespace');
+    listPage.clickCreateYAMLButton();
+    modal.shouldBeOpened();
+    modal.typeInField('input-name', newName);
+    modal.submit();
+    modal.shouldBeClosed();
+    cy.url().should('include', `/k8s/cluster/namespaces/${newName}`);
+
+    cy.log('Updates the Namespace labels');
+    cy.visit('/k8s/cluster/namespaces');
+    listPage.filter.byName(newName);
+    listPage.rows.hasLabel(newName, 'No labels');
+    listPage.rows.clickKebabAction(newName, 'Edit Labels');
+    modal.shouldBeOpened();
+    modal.typeInField('tags-input', labelAppFrontend);
+    modal.submit();
+    modal.shouldBeClosed();
+    listPage.rows.hasLabel(newName, labelAppFrontend);
+  });
+
+  it('Deletes the Namespace', () => {
```

Comment:
This test depends on the previous test having run, so it's not independent. I really want to start off using the best practices, and test independence is important.

https://docs.cypress.io/guides/references/best-practices.html#Having-tests-rely-on-the-state-of-previous-tests

Reply:
fixed